src/app/address/controllers/address.js

```diff
@@ -84,7 +99,34 @@ class AddressController extends BaseController {
       validFrom: addressYearFrom,
     };
 
-    return address;
+    const isChanged = this.checkForChanges(address, chosenAddress);
+    if (isChanged) {
+      delete chosenAddress.uprn;
+    }
+
+    // any changed fields will be overwriten by 'address'
+    // Retains all special fields (sub building name, UPRN etc)
+    return { ...chosenAddress, ...address };
+  }
+
```

Comment:
The check for changes logic, does a simple comparison of the new address created from address view and the address fields from chosen address. If any of the fields have been modified the logic should delete the UPRN as the address could have changed.